### PR TITLE
Add configurable default uplink port (legacy/auto/manual) and selection logic

### LINKED ISSUE #264: Enhancement request for a switch
Hi ... Many thanks for doing the work on these cards .. I am using a selection in my system and find them useful.  Thanks to the [BeardedTinker](https://www.youtube.com/@BeardedTinker) for bringing the integration to my attention.

One observation is that on my USW Flex 2.5G 8 port the card shows "link status - No Link" and speed is blank.  That state correctly reflects the 10Gb RJ45 link port which is selected by default.  I am using the SFP port to link the switch to my DR7 which shows up as port 10 .. I can click the port and it then shows the correct detail but if I refresh the card it switches back to showing the RJ45 link.

As the two ports are mutually exclusive would it be possible to set the SFP port as the default if its active? or maybe include an option in the card ... obviously if its not active you may not be able to determine the default unless its a flag in the card

EG ... when I open the card I see

Image

What I would ideally like to see is

Image

Many thanks

### PR DESCRIPTION
**Pull request by bluenazgul**
### Motivation
- Provide an editor option to control which port is initially selected for Switch/Gateway devices instead of always using the first port. 
- Allow three behaviors: preserve legacy first-port behavior, automatically prefer an active uplink, or manually pick a specific uplink slot. 
- Ensure the card selects a sensible default when device context is loaded and `dynamic_port_details` is not enabled. 

### Description
- Introduces a new config key `default_uplink_port` with editor UI, translations, README sample, and CHANGELOG note to expose the new option (`unset`/`auto`/`<uplink slot key`).
- Adds `getDefaultPort(ports, uplinkPorts, preference, isConnected)` to `src/helpers.js` to encapsulate selection logic for legacy/auto/manual choices.
- Integrates the new helper into the card lifecycle in `src/unifi-device-card.js` to set `_selectedKey` using the configured preference when loading device context or when config changes without dynamic details enabled.
- Updates the editor in `src/unifi-device-card-editor.js` to render the `default_uplink_port` selector, persist the setting, and keep UI state for the details toggle.

### Testing
- Added unit tests in `tests/default-port.mjs` exercising automatic, fallback, manual, and edge-case behaviors and ran them with `node tests/default-port.mjs` which printed "Default port selection tests passed." indicating success.
- No other automated test failures were observed during the rollout.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_6a94139601148333818e7f5f33e90e1a)

### CODE REVIEW

File: src/unifi-device-card.js [hunk 1]
```diff
@@ -1310,7 +1326,12 @@ class UnifiDeviceCard extends HTMLElement {
       if (!selectedStillExists) {
         this._selectedKey = this._config?.dynamic_port_details === true
           ? null
-          : available[0]?.key || null;
+          : getDefaultPort(
+            available,
+            specials,
+            this._config?.default_uplink_port,
+            (slot) => this._isPortConnected(slot)
+          )?.key || null;
```
[the remark this answers is not in the document]
Comment (bluenazgul) [reply src/unifi-device-card.js]: @codex address this issue
